packages/embark-rpc-manager/src/lib/index.ts

```diff
@@ -0,0 +1,84 @@
+import { Callback, Embark, Events, Logger } /* supplied by @types/embark in packages/embark-typings */ from "embark";
+import { AccountParser, dappPath } from "embark-utils";
+import Web3 from "web3";
+import EthAccounts from "./eth_accounts";
+import EthSendTransaction from "./eth_sendTransaction";
+import EthSignTypedData from "./eth_signTypedData";
+import PersonalNewAccount from "./personal_newAccount";
+import BaseTransaction from "./rpcModifier";
+const EmbarkEvents = require("embark-events");
+
+export default class TransactionManager {
+
+  private modifiers: BaseTransaction[] = [];
+  private _web3: Web3 | null = null;
+  private rpcModifierEvents: Events;
+  private logger: Logger;
+  private events: Events;
+  public _accounts: any[] | null = null;
+  public _nodeAccounts: any[] | null = null;
+  constructor(private readonly embark: Embark) {
+    this.events = embark.events;
+    this.logger = embark.logger;
+    this.rpcModifierEvents = new EmbarkEvents();
+    this.init();
+  }
+
+  private async init() {
+    const web3 = await this.web3;
+    const nodeAccounts = await this.nodeAccounts;
+    const accounts = await this.accounts;
+
+    this.rpcModifierEvents.setCommandHandler("nodeAccounts:updated", this.updateAccounts.bind(this));
```

Comment:
feels unnecessary, why not just usee the main event bus properly namespaced?

Reply:
see embark-deployment, it also uses event bus, but doesn't need a new one, just namespaced

Reply:
The reasoning is so that we can use a private instance embark events inside of another package embark-rpc-manager, so as not to pollute the global events (which could incorrectly allow other developers to use events they're not meant to be using - and potentially cause race conditions).